70_FLIPFluids/app_resources/gridUtils.hlsl

```diff
@@ -0,0 +1,67 @@
+#ifndef _FLIP_EXAMPLE_GRID_UTILS_HLSL
+#define _FLIP_EXAMPLE_GRID_UTILS_HLSL
+
+#ifdef __HLSL_VERSION
+struct SGridData
+{
+    float gridCellSize;
+    float gridInvCellSize;
+    float pad0[2];
+
+    int4 particleInitMin;
+    int4 particleInitMax;
+    int4 particleInitSize;
+
+    float4 worldMin;
+    float4 worldMax;
+    int4 gridSize;
```

Comment:
you can use `int32_t3` and `float32_t3` without worry now, scalar layout makes them alignmen to 4 bytes and actually have a size of 12